### ISSUE TITLE
aws lambda cannot be debugged locally in WSL2

### ISSUE THREAD

**Issue by trixoflow** [aws/aws-toolkit-jetbrains#2191]
**Describe the bug**
Lambda runs fine locally, but when I try to debug is stack on "Picked up _JAVA_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=57546 -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true -Xshare:off"

Full log:
```
"C:\Program Files\JetBrains\IntelliJ IDEA 2019.1.2\bin\runnerw64.exe" "C:\Program Files\Amazon\AWSSAMCLI\bin\sam.cmd" local invoke --template C:\w\flowplayer\lamda\HelloLamda\HelloWorldFunction\src\main\.aws-sam\build\template.yaml --event "C:\Users\trixo\AppData\Local\Temp\[Local] App-event6.json" --debug-port 58776
C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\chevron\renderer.py:73: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if scope is 0:
Invoking helloworld.App (java8.al2)
Skip pulling image and use local one: amazon/aws-sam-cli-emulation-image-java8.al2:rapid-1.8.0.

Mounting C:\w\flowplayer\lamda\HelloLamda\HelloWorldFunction\src\main\.aws-sam\build\Function as /var/task:ro,delegated inside runtime container
Picked up _JAVA_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=58776 -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true -Xshare:off
```
I also noticed that there is a docker container created and started, that is not deleted, even when I stop debugging

**To reproduce**
1. Create a lambda using intellij
2. Set the configuration setting 
   a. credentials in aws connection
   b. region in aws connection
   c. any text input (I chose API Gateway)
3. Start debug

**Expected behavior**
To stop at the breaking point inside the RequestHandler

**Screenshots**

**Your Environment**
- OS: Win 10 Pro 10.0.19041 Build 19041
- JetBrains product: Intellij
- JetBrains product version: 2020.2.3
- AWS Toolkit version: 1.20-202
- SAM CLI version: SAM CLI, version 1.8.0
- JVM/Python version: java version "1.8.0_271"  Python 3.9.0 

**Additional context**
-

**Comment by abrooksv** [comment]
I do not believe that the WSL2 opened networked ports are visible to Win10 by default.

Are you able to telnet to that debug port from Windows side?

**Comment by trixoflow** [comment]
hi! thank you for the lighting speed answer. Great idea but didn't work :'(

I opened the port 59775 and I executed the following
`sam local invoke --template C:\w\flowplayer\lamda\HelloLamda\HelloWorldFunction\src\main\.aws-sam\build\template.yaml --event "C:\Users\trixo\AppData\Local\Temp\[Local] App-event12665.json" --debug-port 59775`

I got the usual:
```
C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\chevron\renderer.py:73: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if scope is 0:
Invoking helloworld.App (java8.al2)
Skip pulling image and use local one: amazon/aws-sam-cli-emulation-image-java8.al2:rapid-1.8.0.

Mounting C:\w\flowplayer\lamda\HelloLamda\HelloWorldFunction\src\main\.aws-sam\build\Function as /var/task:ro,delegated inside runtime container
Picked up _JAVA_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=59775 -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true -Xshare:off

Then I telnet from windows: "telnet localhost 59775" and I got the following on the other terminal:
`Debugger failed to attach: timeout during handshake`
and it continue to be stack.

**Comment by abrooksv** [comment]
And telnet was able to connect?

**Comment by abrooksv** [comment]
What is the debug host under the SAM CLI tab in the debug run config? That host is populated by the configured docker endpoints in the IDE

**Comment by trixoflow** [comment]
Yes the telnet succeeded sorry for not mention it clearly. I have uploaded all three tabs in the configuration
![2020-11-12 10_16_50-Run_Debug Configurations](https://user-images.githubusercontent.com/72970459/98913675-5b63d980-24d0-11eb-8ffe-0ff2f229790b.png)
![2020-11-12 10_17_04-Run_Debug Configurations](https://user-images.githubusercontent.com/72970459/98913678-5c950680-24d0-11eb-8960-10214c7dfd4d.png)
![2020-11-12 10_17_09-Run_Debug Configurations](https://user-images.githubusercontent.com/72970459/98913681-5dc63380-24d0-11eb-8585-ef5902db1623.png)

**Comment by trixoflow** [comment]
This looks like this bug: https://github.com/aws/aws-toolkit-eclipse/issues/164

I tried invoking it from terminal, run passes debug freezes. So probably it is not an aws-toolkit-jetbrains issues, but something more generic, should we move this issues on another repository?

**Comment by amileris** [comment]
What I think happens is that when AWS Toolkit executes `sam.cmd local invoke ...` - it starts up the Docker container (in my case, of `public.ecr.aws/lambda/java:17-rapid-x86_64` image), and then during the container kick-off at some point after the `Picked up _JAVA_OPTIONS...` message it expects a connection from a remote debugger, but not receiving one.

To me it looks like the AWS Toolkit fails to match the debug port provided in `--debug-port` argument to `sam.cmd local invoke ...` when creating a debugger connection. I've tested this by manually creating a `Remote JVM Debug` configuration in IntelliJ on the matching port, and was able to continue with debugging of my Lambda code.

![image](https://github.com/aws/aws-toolkit-jetbrains/assets/25879679/4b2f07c2-a8bb-4524-af7f-babed1afff96)
